modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -412,7 +412,9 @@ private Sku _updateSKU(CPInstance cpInstance, Sku sku) throws Exception {
 		int expirationDateHour = 0;
 		int expirationDateMinute = 0;
 
-		if (cpInstance.getExpirationDate() != null) {
+		if ((cpInstance.getExpirationDate() != null) ||
+			(sku.getExpirationDate() != null)) {
+
```

Comment:
if (sku.getExpirationDate() != null) { , test with this. Only this should be good enough.